tests/receivers/discovery/host_observer_test.go

```diff
@@ -17,18 +17,92 @@
 package tests
 
 import (
+	"fmt"
+	"path/filepath"
 	"testing"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/plogtest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/signalfx/splunk-otel-collector/tests/testutils"
 )
 
 func TestDiscoveryReceiverWithHostObserverAndSimplePrometheusReceiverProvideStatusLogs(t *testing.T) {
 	testutils.SkipIfNotContainerTest(t)
-	if testutils.CollectorImageIsForArm(t) {
-		t.Skip("host_observer missing process info on arm")
-	}
-	testutils.AssertAllLogsReceived(
-		t, "host_observer_simple_prometheus_statuses.yaml",
-		"host_observer_simple_prometheus_config.yaml", nil, nil,
-	)
+	tc := testutils.NewTestcase(t)
+	defer tc.PrintLogsOnFailure()
+	defer tc.ShutdownOTLPReceiverSink()
+
+	_, shutdown := tc.SplunkOtelCollector("host_observer_simple_prometheus_config.yaml")
+	defer shutdown()
+
+	failure, err := golden.ReadLogs(filepath.Join("testdata", "expected", "host_observer.yaml"))
+	require.NoError(t, err)
+	success, err := golden.ReadLogs(filepath.Join("testdata", "expected", "host_observer-connected.yaml"))
```

Comment:
Why can't we keep them in the same file? I believe they are supposed to be sent in one batch.

Reply:
they are not sent in one batch, no. I think we don't use the batch processor as well.

Reply:
here, Dmitrii defines one batch as one payload of data under pdata.Logs. That means one plog.Logs, containing one plog.ResourceLogs, containing one plog.ScopeLogs, containing two plog.LogRecords.